config/main.py

```diff
@@ -1852,27 +1855,71 @@ def override_config_table(db, input_config_db, dry_run):
 
     config_db = db.cfgdb
 
+    # Read config from configDB
+    current_config = config_db.get_config()
+    # Serialize to the same format as json input
+    sonic_cfggen.FormatConverter.to_serialized(current_config)
+
+    updated_config = update_config(current_config, config_input)
+
+    yang_enabled = is_yang_config_verification_enabled(config_db)
+    if yang_enabled:
+        sy = sonic_yang_with_loaded_models(YANG_DIR)
+        # Validate running config
+        validate_config_by_yang(sy, current_config)
+        # Validate input config
+        validate_config_by_yang(sy, config_input)
```

Comment:
Why do we need to validate the input config and also the updated whole config? Does updated whole config not include input config?

Reply:
It is an enforcement to input config. The assumption is that input config should be self-contained, and do not need anything more in current_config on device.

With this assumption, we could more easily to tell a process failure is due to invalid input or sonic config error or logic bug.

Reply:
In that case, should we add some logging to `validate_config_by_yang()` (or in between calls) so we know which config caused the error?

Reply:
Agree with @isabelmsft, after adding this new logging, also need test the logging.

Reply:
Added.